moment-range/index.d.ts

```diff
@@ -1,46 +1,57 @@
-// Type definitions for Moment.js 2.0.4
+// Type definitions for Moment-range.js 3.0.3
 // Project: https://github.com/gf3/moment-range
-// Definitions by: Bart van den Burg <https://github.com/Burgov>, Wilgert Velinga <https://github.com/wilgert>
+// Definitions by: Bart van den Burg <https://github.com/Burgov>, Wilgert Velinga <https://github.com/wilgert>, Juan Francisco Adame <https://github.com/franjuan>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 import * as moment from 'moment';
 
 export = moment;
 
+type interval = 'years' | 'quarters' | 'months' | 'weeks' | 'days' | 'hours' | 'minutes' | 'seconds' | 'milliseconds';
+
 declare module 'moment' {
      interface Range {
         start: Moment;
         end: Moment;
 
-        contains (other: Date, exclusive?: boolean): boolean;
-        contains (other: Moment, exclusive?: boolean): boolean;
+        contains (other: Range, options?: {exclusive?: boolean}): boolean;
 
-        overlaps (range: Range): boolean;
+        overlaps (range: Range, options?: {adjacent?: boolean}): boolean;
 
-        intersect (other: Range): Range;
+        intersect (other: Range, options?: {adjacent?: boolean}): Range;
 
         add (other: Range): Range;
 
         subtract (other: Range): Range[];
 
-        by (range: string, hollaback: (current: Moment) => void, exclusive?: boolean): void;
-        by (range: Range, hollaback: (current: Moment) => void, exclusive?: boolean): void;
+        by(interval: interval, options?: {exclusive?: boolean, step?: number}): Iterable<Range>;
+
+        byRange(interval: Range, options?: {exclusive?: boolean, step?: number}): Iterable<Range>;
 
         isSame (other: Range): boolean;
 
-        diff (unit?: string): number;
+        diff (unit?: unitOfTime.Diff, precise?: boolean): number;
 
-        toDate (): Date;
+        toDate (): Date[];
 
         toString (): string;
 
-        toArray (by: Range|String, exclusive?: boolean): Moment[];
-
         valueOf (): number;
 
-        center (): number;
+        center (): Moment;
 
         clone (): Range;
+
+        isEqual (other: Range): boolean;
+
+        adjacent (other: Range): boolean;
+
+        duration (unit?: unitOfTime.Diff, precise?: boolean): number;
+
+        reverseBy(interval: interval, options?: {exclusive?: boolean, step?: number}): Iterable<Range>;
```

Comment:
these should return an `Iterable<Moment>`, see https://github.com/gf3/moment-range/blob/b500d6a15dc1149a6cc3f5c941d4d410a18a0bc6/README.md#reversebyrange